### PR TITLE
WP-17438 Error handling clean up

### PR DESCRIPTION
**Pull request by jayma91**
WP-17438 Error handling clean up
https://varicent.atlassian.net/browse/WP-17438

notes:
- added API error handling

### CODE REVIEW

File: tap_hubspot/__init__.py [hunk 1]
```diff
@@ -228,10 +228,18 @@ def acquire_access_token_from_refresh_token():
...
 
     resp = requests.post(BASE_URL + "/oauth/v1/token", data=payload)
-    if resp.status_code == 403:
-        raise SymonException(f'Failed to connect to Hubspot, please ensure the oauth token is update to date.', 'hubspot.AuthInvalid')
 
-    resp.raise_for_status()
+    try:
+        resp.raise_for_status()
+    except requests.exceptions.HTTPError as e:
+        if e.response is not None and e.response.status_code == 400:
```
Comment (jayma91) [inline tap_hubspot/__init__.py:235]: status code 400 used for invalid creds.

File: tap_hubspot/__init__.py [hunk 2]
```diff
@@ -228,10 +228,23 @@ def acquire_access_token_from_refresh_token():
 
 
     resp = requests.post(BASE_URL + "/oauth/v1/token", data=payload)
-    if resp.status_code == 403:
-        raise SymonException(f'Failed to connect to Hubspot, please ensure the oauth token is update to date.', 'hubspot.AuthInvalid')
+    if resp.status_code == 400:
+        raise SymonException(f'Failed to connect to Hubspot. Please ensure the OAuth token is up to date.', 'hubspot.AuthInvalid')
```
Comment (jayma91) [inline tap_hubspot/__init__.py:232]: status code 400 was used for invalid token.

File: tap_hubspot/__init__.py [hunk 3]
```diff
@@ -228,10 +228,23 @@ def acquire_access_token_from_refresh_token():
...
+        resp.raise_for_status()
+    except requests.exceptions.HTTPError as e:
+        message = None
+        try:
+            message = resp.json()["message"]
+        except Exception:
+            pass
+
+        if message is not None:
+            raise SymonException(f'Import failed with the following Hubstpot error: {message}', 'hubspot.HubspotApiError')
```
Comment (jayma91) [inline tap_hubspot/__init__.py:244]: example: 
```
tap_hubspot.symon_exception.SymonException: Import failed with the following Hubstpot error: (status code: 401) Authentication credentials not found. This API supports OAuth 2.0 authentication and you can find more details at https://developers.hubspot.com/docs/methods/auth/oauth-overview
```
error doc: https://legacydocs.hubspot.com/docs/faq/api-error-responses

File: tap_hubspot/__init__.py [hunk 4]
```diff
@@ -304,10 +321,19 @@ def request(url, params=None):
...
             resp.raise_for_status()
+        except requests.exceptions.HTTPError as e:
+            message = None
+            try:
+                message = resp.json()["message"]
+            except:
+                pass
+
+            if message is not None:
+                raise SymonException(f'Import failed with the following Hubstpot error: {message}', 'hubspot.HubspotApiError')
```
Comment (mjdoor) [inline tap_hubspot/__init__.py:335]: Typo on Hubspot in a few places: Hubstpot

**Review by mjdoor — APPROVED** [review]

**Pull request merged**